pkg/thing/thing.go

```diff
@@ -312,6 +368,42 @@ func (b *baseBuilder) Create() (*Thing, error) {
 	if err != nil {
 		return nil, err
 	}
+	if b.authHandler != nil {
+		// check we have a signer
+		if b.authHandler.key == nil {
+			return nil, fmt.Errorf("no key to authenticate thing with")
+		}
+		info, err := client.amInfo()
+		if err != nil {
+			return nil, err
+		}
+		auth := callback.AuthenticateHandler{
+			Realm:   info.Realm,
+			ThingID: b.authHandler.thingID,
+			KeyID:   b.authHandler.keyID,
+			Key:     b.authHandler.key,
+			Claims:  b.authHandler.claims,
+		}
+		if b.regHandler != nil {
+			if auth.KeyID == "" {
```

Comment:
This ID generation seems a bit hidden. I think we should make the KeyID mandatory, so fail if it's empty. Along with exporting the thumbprint function, it should be clear what is expected.

Reply:
Good point 👍